apps/fabric-website/src/components/Site/Site.tsx

```diff
@@ -249,12 +251,14 @@ export class Site<TPlatforms extends string = string> extends React.Component<IS
   }
 
   private _renderPageNav(): JSX.Element | null {
-    const { activePages, searchablePageTitle, isContentFullBleed, pagePlatforms = {} } = this.state;
+    const { activePages, searchablePageTitle, isContentFullBleed, pagePlatforms } = this.state;
+    const { siteDefinition } = this.props;
 
     if (!isContentFullBleed && activePages) {
       // If current page doesn't have pages for the active platform, switch to its first platform.
       if (Object.keys(pagePlatforms).length > 0 && activePages.length === 0) {
-        this._onPlatformChanged(Object.keys(pagePlatforms)[0] as TPlatforms);
+        const firstPlatform = getPageFirstPlatform(getSiteArea(siteDefinition.pages), siteDefinition);
```

Comment:
Thinking about it more, this logic probably belongs somewhere else, since render functions in theory shouldn't have side effects (especially not a side effect as dramatic as this). Also, the state updates made by this call won't have taken effect yet by the time `this._renderPlatformPicker()` is called, resulting in the platform picker temporarily rendering incorrectly. That will be fixed after the re-render caused by the state updates in `this._onPlatformChanged()`, but it would be better to have this update somewhere else and skip the extra re-renders.

Similarly to the `initialRender`/`_getNavData()` changes discussed in another comment thread, I'm fine with checking this in as-is for now (it's already fixing a major bug!) and moving things around later.